### ISSUE TITLE
CIFAR10: model predicts random coloured images

### ISSUE THREAD

**Issue by lhaippp** [lucidrains/denoising-diffusion-pytorch#134]
Hi there, thanks for your excellent work!

I'm trying DDPM with cifar10 dataset, it seems that the loss converges well... (from 0.8 to 0.04)
However, the network always predicts random colored images, for example
![ezgif com-gif-maker (3)](https://user-images.githubusercontent.com/22257640/205317201-a4d0bd65-092b-4227-9961-633adb83cafc.gif)

below are my scripts
```
import cv2
import torch

import numpy as np

from denoising_diffusion_pytorch import Unet, GaussianDiffusion, Trainer

model = Unet(dim=64, dim_mults=(1, 2, 4, 8)).cuda()

diffusion = GaussianDiffusion(
    model,
    image_size=128,
    timesteps=1000,  # number of steps
    sampling_timesteps=250,  # number of sampling timesteps (using ddim for faster inference [see citation for ddim paper])
    loss_type='l1',  # L1 or L2
    p2_loss_weight_gamma=1.,
).cuda()

trainer = Trainer(
    diffusion,
    '/data/denoising-diffusion-pytorch/dataset/cifar-10-batches-py',
    train_batch_size=16,
    train_lr=1e-4,
    train_num_steps=70000,  # total training steps
    gradient_accumulate_every=2,  # gradient accumulation steps
    ema_decay=0.995,  # exponential moving average decay
    amp=False,  # turn on mixed precision
)

trainer.train()
```
does anyone have the same situation? thanks for your help in advance!

**Comment by lhaippp** [comment]
I set `p2_loss_weight_gamma=1.,` to `p2_loss_weight_gamma=0,` which solved my problem...

**Comment by pimakshay** [comment]
@lhaippp I don't find the `p2_loss_weight_gamma` in `denoising_diffusion_pytorch.py`. Could you please let me know what this parameter is and what changes you made in the code?
